File: backend/test-rds-connection.js
const { Sequelize } = require('sequelize');

async function testRDSConnection() {
  console.log('=== 测试阿里云RDS数据库连接 ===');
  
  const sequelize = new Sequelize('cyrg2025', 'hotdog', 'Zhkj@62102218', {
    host: process.env.DB_HOST || 'rm-uf660d00xovkm30678o.sqlserver.rds.aliyuncs.com',
    port: parseInt(process.env.DB_PORT) || 1433,
    dialect: 'mssql',
    logging: false,
    dialectOptions: {
      options: {
        encrypt: false,
        trustServerCertificate: true
      }
    },
    timezone: '+08:00'
  });

  try {
    console.log('正在连接数据库...');
    await sequelize.authenticate();
    console.log('✅ 数据库连接成功！');
    
    // 测试查询
    console.log('正在测试查询...');
    const [results] = await sequelize.query('SELECT @@VERSION as version');
    console.log('✅ 查询测试成功！');
    console.log('SQL Server版本:', results[0].version);
    
    // 检查数据库是否存在
    console.log('正在检查数据库...');
    const [databases] = await sequelize.query("SELECT name FROM sys.databases WHERE name = 'cyrg2025'");
    if (databases.length > 0) {
      console.log('✅ 数据库 cyrg2025 存在');
    } else {
      console.log('⚠️  数据库 cyrg2025 不存在，需要创建');
    }
    
  } catch (error) {
    console.error('❌ 数据库连接失败:', error.message);
    console.error('错误详情:', error);
  } finally {
    await sequelize.close();
    console.log('数据库连接已关闭');
  }
}

testRDSConnection(); 

File: archive/scripts/test_rds_connection.py
#!/usr/bin/env python3
"""
测试RDS数据库连接
"""
import pymssql

def test_rds_connection():
    """测试RDS连接"""
    try:
        # RDS连接配置
        conn = pymssql.connect(
            server='rm-uf660d00xovkm30678o.sqlserver.rds.aliyuncs.com',
            port=1433,
            user='hotdog',
            password='Zhkj@62102218',
            database='hotdog2030'
        )
        
        cursor = conn.cursor()
        cursor.execute("SELECT @@VERSION")
        version = cursor.fetchone()[0]
        print(f"✅ RDS连接成功!")
        print(f"数据库版本: {version[:100]}...")
        
        # 检查数据库列表
        cursor.execute("SELECT name FROM sys.databases WHERE name IN ('hotdog2030', 'cyrg2025')")
        databases = cursor.fetchall()
        print(f"可用数据库: {[db[0] for db in databases]}")
        
        conn.close()
        return True
        
    except Exception as e:
        print(f"❌ RDS连接失败: {e}")
        return False

if __name__ == "__main__":
    test_rds_connection()
